imdb_clone/settings.py

```diff
@@ -0,0 +1,168 @@
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "django-insecure-2j-+jb@$#m1dgxz^(kjs-9!!56rmmuf&$7oh2ma7qaqv@=u+w@"
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = ["*"]
+
+CORS_ALLOWED_ORIGINS = ["http://localhost:3000" "http://127.0.0.1:3000"]
+
+CORS_ALLOWED_CREDENTIALS = True
+
+CORS_ORIGIN_ALLOW_ALL = True
+
+# Application definition
+
+INSTALLED_APPS = [
+    "django.contrib.admin",
+    "django.contrib.auth",
+    "django.contrib.contenttypes",
+    "django.contrib.sessions",
+    "django.contrib.messages",
+    "django.contrib.staticfiles",
+    "imdb_django",
+    "rest_framework",
+    "rest_framework_simplejwt",
+    "user_api_app",
+    "django_filters",
+]
+
+MIDDLEWARE = [
+    "corsheaders.middleware.CorsMiddleware",
+    "django.middleware.security.SecurityMiddleware",
+    "django.contrib.sessions.middleware.SessionMiddleware",
+    "django.middleware.common.CommonMiddleware",
+    "django.middleware.csrf.CsrfViewMiddleware",
+    "django.contrib.auth.middleware.AuthenticationMiddleware",
+    "django.contrib.messages.middleware.MessageMiddleware",
+    "django.middleware.clickjacking.XFrameOptionsMiddleware",
+]
+
+ROOT_URLCONF = "imdb_clone.urls"
+
+TEMPLATES = [
+    {
+        "BACKEND": "django.template.backends.django.DjangoTemplates",
+        "DIRS": [],
+        "APP_DIRS": True,
+        "OPTIONS": {
+            "context_processors": [
+                "django.template.context_processors.debug",
+                "django.template.context_processors.request",
+                "django.contrib.auth.context_processors.auth",
+                "django.contrib.messages.context_processors.messages",
+            ],
+        },
+    },
+]
+
+
+REST_FRAMEWORK = {
+    "DEFAULT_AUTHENTICATION_CLASSES": (
+        "rest_framework_simplejwt.authentication.JWTAuthentication",
+    ),
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.PageNumberPagination",
+    "PAGE_SIZE": 100,
+}
+
+
+import datetime
+
+SIMPLE_JWT = {
+    "ACCESS_TOKEN_LIFETIME": datetime.timedelta(hours=1),  # Adjust as needed
+    "SLIDING_TOKEN_REFRESH_LIFETIME": datetime.timedelta(days=7),
+    "SLIDING_TOKEN_LIFETIME": datetime.timedelta(days=14),
+    "SLIDING_TOKEN_REFRESH_GRACE_PERIOD": datetime.timedelta(days=1),
+    "SLIDING_TOKEN_VERIFYING_PERIOD": datetime.timedelta(days=30),
+}
+
+
+AUTHENTICATION_BACKENDS = [
+    "user_api_app.custom_auth_backends.EmailBackend",
+    "django.contrib.auth.backends.ModelBackend",
+]
+
+
+WSGI_APPLICATION = "imdb_clone.wsgi.application"
+
+
+# Database
+# https://docs.djangoproject.com/en/4.2/ref/settings/#databases
+
+DATABASES = {
+    "default": {
+        "ENGINE": "django.db.backends.mysql",
+        "NAME": "imdb",
+        "USER": "root",
+        "PASSWORD": "Toorjr-16",
```

Comment:
Ideally, we should not keep passwords here (db_name, user, host, port, etc), we can use an env file. For now, we are good, but go through the following link so that you have knowledge of this

https://dev.to/defidelity/protect-your-sensitive-data-a-guide-to-env-files-in-django-499e